internal/entities/events.go

```diff
@@ -0,0 +1,74 @@
+package entities
+
+type EventType string
+
+type Event interface {
+	Type() EventType
+}
+
+const (
+	EventTypeUpdateTree        EventType = "update tree"
+	EventTypeCreateTree        EventType = "create tree"
+	EventTypeDeleteTree        EventType = "delete tree"
+	EventTypeUpdateTreeCluster EventType = "update tree cluster"
```

Comment:
Do we only have to listen on Update for a tree cluster?

Reply:
No, but I would implement each event if it is really needed